projects/gateway2/translator/plugins/httplisteneroptions/query/query.go

```diff
@@ -60,43 +65,68 @@ func (q *queries) GetAttachedHttpListenerOptions(
 	ctx context.Context,
 	listener *gwv1.Listener,
 	parentGw *gwv1.Gateway,
+	parentListenerSet *apixv1a1.XListenerSet,
 ) ([]*solokubev1.HttpListenerOption, error) {
 	if parentGw == nil {
 		return nil, errors.New("nil parent gateway")
 	}
 	if parentGw.GetName() == "" || parentGw.GetNamespace() == "" {
 		return nil, fmt.Errorf("parent gateway must have name and namespace; received name: %s, namespace: %s", parentGw.GetName(), parentGw.GetNamespace())
 	}
+
+	parentListenerSetName := ""
+	if parentListenerSet != nil {
+		parentListenerSetName = parentListenerSet.GetName()
+	}
+
 	nn := types.NamespacedName{
 		Namespace: parentGw.Namespace,
 		Name:      parentGw.Name,
 	}
-	list := &solokubev1.HttpListenerOptionList{}
+
+	nnListenerSet := types.NamespacedName{
+		Namespace: parentGw.Namespace,
```

Comment:
The listenerset is not always in the same ns as the parent gateway.
```suggestion
		Namespace: parentListenerSet.GetNamespace(),
```